src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -121,55 +121,77 @@ namespace AppInstaller::Repository
 
         std::string GetMappedInstalledVersion(const std::string& installedVersion, const std::shared_ptr<IPackage>& availablePackage)
         {
-            // Construct a map between manifest version and arp version range. The map is ordered in descending by package version.
-            std::vector<std::pair<std::string, Utility::VersionRange>> arpVersionMap;
+            // Stores raw versions value strings to run a preliminary check whether version mapping is needed.
+            std::vector<std::tuple<std::string, std::string, std::string>> rawVersionValues;
             auto versionKeys = availablePackage->GetAvailableVersionKeys();

```

Comment:
I've just realized that if the available package is a composite of multiple sources, then this is flawed.  Do we do that now or not?

Reply:
I think composite available package may need more thinking, currently we stop when we find one from a source, I'll add a comment as a reminder when we enable composite available package one day